scripts/migration/src/v5/merge_config_in_solution_yaml.js

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+const { join } = require('path');
+const fs = require('fs');
+const { getConfigFolder, getOutputFolder } = require('./project.js');
+const { copyFileToMJS, parseESFile } = require('../common/js_modules.js');
+const yaml = require('../common/yaml.js');
+
+const createOutputFolder = () => {
+  fs.mkdirSync(getOutputFolder(), { recursive: true });
+};
+
+const clearFileFromOutputFolder = (fileName) => {
+  const filePath = join(getOutputFolder(), fileName);
+  if (fs.existsSync(filePath)) fs.unlinkSync(filePath);
+};
+
+const copyConfigFileToMJS = (fileName) => {
+  return copyFileToMJS(getConfigFolder(), fileName, getOutputFolder());
+};
+
+const preventRolesImportInFile = (filePath) => {
+  const rolesReplaceMapping = {
+    'APP_ROLES.OrganizationAdmin': "'Organization.Admin'",
+    'APP_ROLES.OrganizationCollaborator': "'Organization.Collaborator'",
+    'APP_ROLES.OrganizationModeler': "'Organization.Modeler'",
+    'APP_ROLES.OrganizationUser': "'Organization.User'",
+    'APP_ROLES.OrganizationViewer': "'Organization.Viewer'",
+    'APP_ROLES.PlatformAdmin': "'Platform.Admin'",
+  };
+
+  const contentStr = fs.readFileSync(filePath, { encoding: 'utf8', flag: 'r' });
+  const contentArray = contentStr.split('\n');
+  // Remove line importing the APP_ROLES constant if it exists
+  const patterns = ['import', 'APP_ROLES'];
+  const isRolesImportLine = (line) => patterns.every((pattern) => line.includes(pattern));
+  const importLineIndex = contentArray.findIndex((line) => isRolesImportLine(line));
+  contentArray.splice(importLineIndex, 1);
+  // Replace mentions of APP_ROLES object
+  const newContentArray = contentArray.map((line) => {
+    let newLine = line;
+    for (const [roleKey, roleValue] of Object.entries(rolesReplaceMapping)) {
+      newLine = newLine.replaceAll(roleKey, roleValue);
+    }
+    return newLine;
+  });
+  // Overwrite content of the input file
+  fs.writeFileSync(filePath, newContentArray.join('\n'));
+};
+
+const parseScenarioParametersConfigFile = async () => {
+  const mjsFilePath = copyConfigFileToMJS('ScenarioParameters.js');
+  preventRolesImportInFile(mjsFilePath);
+  const scenarioParametersConfig = await parseESFile(mjsFilePath);
+  clearFileFromOutputFolder('ScenarioParameters.mjs');
+  return scenarioParametersConfig;
+};
+
+const reshapeObjectToArrayOfObjects = (object) => {
+  const arrayOfObjects = [];
+  for (const [key, value] of Object.entries(object)) {
```

Comment:
Actually, the syntax would be `forEach(([key, value]) => { ... });` , I'm not sure it would be easier to write or read, but I would be glad to discuss it in a dev meeting if you want :smile: